tests/planner/parabolic/test_ParabolicSmoother.cpp

```diff
@@ -0,0 +1,231 @@
+#include <gtest/gtest.h>
+#include <aikido/statespace/Rn.hpp>
+#include <aikido/statespace/GeodesicInterpolator.hpp>
+#include <aikido/statespace/SO2.hpp>
+#include <aikido/statespace/SO3.hpp>
+#include <aikido/statespace/CartesianProduct.hpp>
+#include <aikido/planner/parabolic/ParabolicTimer.hpp>
+#include <aikido/planner/parabolic/ParabolicSmoother.hpp>
+#include <aikido/constraint/Satisfied.hpp>
+#include <aikido/util/StepSequence.hpp>
+#include "eigen_tests.hpp"
+
+using Eigen::Vector2d;
+using Eigen::Vector3d;
+using aikido::trajectory::Interpolated;
+using aikido::statespace::GeodesicInterpolator;
+using aikido::statespace::R2;
+using aikido::statespace::CartesianProduct;
+using aikido::statespace::SO2;
+using aikido::statespace::SO3;
+using aikido::statespace::StateSpacePtr;
+using aikido::constraint::Satisfied;
+using aikido::planner::parabolic::convertToSpline;
+using aikido::planner::parabolic::doShortcut;
+using aikido::planner::parabolic::doBlend;
+using aikido::planner::parabolic::doShortcutAndBlend;
+
+class ParabolicSmootherTests : public ::testing::Test
+{
+public:
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+protected:
+  void SetUp() override
+  {
+    mRng = aikido::util::RNGWrapper<std::mt19937>( std::random_device{}() );
+    mStateSpace = std::make_shared<R2>();
+    mMaxVelocity = Eigen::Vector2d(20., 20.);
+    mMaxAcceleration = Eigen::Vector2d(10., 10.);
+
+    mInterpolator = std::make_shared<GeodesicInterpolator>(mStateSpace);
+    mStraightLine = std::make_shared<Interpolated>(
+      mStateSpace, mInterpolator);
+
+    auto state = mStateSpace->createState();
+
+    state.setValue(Vector2d(1., 2.));
+    mStraightLine->addWaypoint(0., state);
+
+    state.setValue(Vector2d(3., 4.));
+    mStraightLine->addWaypoint(1., state);
+
+    mNonStraightLineLength = initNonStraightLine();
```

Comment:
It's odd to have one output (`mNonStraightLine`) in a member variable and a second `mNonStraightLineLength` as a return value. Please stick to one convention or the other for this function.